app/models/sources/fog.rb

```diff
@@ -64,32 +64,45 @@ def decommission_proxy(proxy)
     # Adds them to the site if provided
     # Parameters:
     #   site - What site to add the found servers to (if any)
-    def find_orphaned_servers!(*args)
-      num_servers_found = 0
+    #   desired_proxy_count - Until the site owns this many proxies, add new
+    #     proxies to site
+    def find_orphaned_servers!(
+      site: NoSite,
+      desired_proxy_count: self.max_proxies
+    )
+      total_known_proxies = proxies.active.length
+      servers_still_building = false
       connection.servers.each do |server|
-        if server_is_proxy_type?(server) \
-            && server.ready? \
-            && !Proxy.active.where(:host => server.public_ip_address).exists?
-
-          # An "orphaned" server can be in the middle of a decommission.
-          # We still need to count it so that we don't try to create more
-          # servers than the source's max.
-          unless recent_decommissions.include? server.name
-            save_server server, *args
-            Activity << "Found orphaned proxy #{server.public_ip_address} (#{server.name}) for site(s) #{args.map(&:name).join ', '}"
+        if server_is_proxy_type?(server)
+          if !server.ready?
+            servers_still_building = true
+          elsif !Proxy.active.where(:host => server.public_ip_address).exists? \
+            && !recent_decommissions.include?(server.name)
+
+            save_server server, site
+
+            msg = "Found orphaned proxy #{server.public_ip_address} (#{server.name})"
+            msg += " for site #{site.name}" if site.respond_to? 'name'
+            Activity << msg
+
+            total_known_proxies += 1
+            # Don't add found proxies to the site if we've already reached our
+            # quota
+            site = NoSite if total_known_proxies >= desired_proxy_count
```

Comment:
Why is this happening here? If `total_known_proxies > desired_proxy_count`, we'll already have added the proxy to the site on line 82. Seems like this check should happen before the call to `#save_server`.